src/pycrunch/cubes.py

```diff
@@ -35,36 +38,48 @@ def fetch_cube(dataset, dimensions, weight=None, filter=None, **measures):
 
     """
     dims = []
-    for d in dimensions:
-        if isinstance(d, dict):
+    variables_alias = dataset.variables.by('alias')
+    variables_name = dataset.variables.by('name')
+    for dim in dimensions:
+        if isinstance(dim, dict):
             # This is already a Crunch expression.
-            dims.append(d)
-        elif isinstance(d, six.string_types):
-            ref = {'variable': d}
+            dims.append(dim)
+        elif isinstance(dim, six.string_types):
             # A URL of a variable entity. GET it to find its type.
-            v = dataset.session.get(d).payload
-            if v.body.type == "numeric":
+            try:
+                var = dataset.session.get(dim).payload
```

Comment:
This seems a bit expensive to do inside a forloop, do you need the full variable resource here?

Are the attributes you will use about the variable available in the variable catalog?

Reply:
I don't really know. They could be, and they probably are. I'm creating a new branch, so it can go through 🎫 system, and will continue the communication there.